x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -2,17 +2,146 @@ package keeper
+	shannonAccAddr := cosmostypes.MustAccAddressFromBech32(msg.ShannonDestAddress)
+
+	// Ensure that a MorseClaimableAccount exists for the given morseSrcAddress.
+	morseClaimableAccount, isFound := k.GetMorseClaimableAccount(
+		sdkCtx,
+		msg.MorseSrcAddress,
+	)
+	if !isFound {
+		return nil, status.Error(
+			codes.NotFound,
+			migrationtypes.ErrMorseSupplierClaim.Wrapf(
+				"no morse claimable account exists with address %q",
+				msg.MorseSrcAddress,
+			).Error(),
+		)
+	}
+
+	// Ensure that the given MorseClaimableAccount has not already been claimed.
+	if morseClaimableAccount.IsClaimed() {
+		return nil, status.Error(
+			codes.FailedPrecondition,
+			migrationtypes.ErrMorseSupplierClaim.Wrapf(
+				"morse address %q has already been claimed at height %d by shannon address %q",
+				msg.MorseSrcAddress,
+				morseClaimableAccount.ClaimedAtHeight,
+				morseClaimableAccount.ShannonDestAddress,
+			).Error(),
+		)
+	}
+
+	// Default to the supplier stake amount recorded in the MorseClaimableAccount.
+	if msg.Stake == nil {
+		msg.Stake = &morseClaimableAccount.SupplierStake
+	}
+
+	// Mint the totalTokens to the shannonDestAddress account balance.
+	// The Supplier stake is subsequently escrowed from the shannonDestAddress account balance.
+	// NOTE: The current supplier module's staking fee parameter will subsequently be deducted
+	// from the claimed balance.
+	if err := k.MintClaimedMorseTokens(ctx, shannonAccAddr, morseClaimableAccount.TotalTokens()); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// Set ShannonDestAddress & ClaimedAtHeight (claim).
+	morseClaimableAccount.ShannonDestAddress = shannonAccAddr.String()
+	morseClaimableAccount.ClaimedAtHeight = sdkCtx.BlockHeight()
+
+	// Update the MorseClaimableAccount.
+	k.SetMorseClaimableAccount(
+		sdkCtx,
+		morseClaimableAccount,
+	)
+
+	msgStakeSupplier := suppliertypes.NewMsgStakeSupplier(
+		shannonAccAddr.String(),
+		shannonAccAddr.String(),
+		shannonAccAddr.String(),
+		*msg.Stake,
+		[]*sharedtypes.SupplierServiceConfig{msg.ServiceConfig},
+	)
+
+	initialSupplierStake := cosmostypes.NewCoin(volatile.DenomuPOKT, math.ZeroInt())
+	foundSupplier, isFound := k.supplierKeeper.GetSupplier(ctx, shannonAccAddr.String())
+	if isFound {
+		initialSupplierStake = *foundSupplier.Stake
+	}
+
+	supplier, err := k.supplierKeeper.StakeSupplier(ctx, logger, msgStakeSupplier)
+	if err != nil {
+		// DEV_NOTE: StakeSupplier SHOULD ALWAYS return a gRPC status error.
+		return nil, err
+	}
+
+	// DEV_NOTE: When BOTH:
+	// - the claimed Shannon account is already staked as a supplier
+	// - the MsgClaimMorseSupplier stake amount ("default" or otherwise)
+	//   is less than the current supplier stake amount
+	// then, claimedSupplierStake is set to zero as it would otherwise result in a negative amount.
+	// This value is only used in event(s) and the msg response.
+	claimedSupplierStake, err := supplier.Stake.SafeSub(initialSupplierStake)
```

Comment:
I don't fully understand the need for any arithmetic here.

The claim should simply be what we are claiming from Morse.